tools/azure-sdk-tools/ci_tools/dependency_analysis.py

```diff
@@ -0,0 +1,352 @@
+#!/usr/bin/env python
+from __future__ import print_function, unicode_literals
+import argparse
+import ast
+from datetime import datetime
+import glob
+import io
+import json
+import os
+import re
+import sys
+import textwrap
+from typing import List, Set, Dict, Tuple, Any
+from collections import Sized
+
+from pkg_resources import Requirement
+from packaging.specifiers import SpecifierSet, Version
+from ci_tools.variables import discover_repo_root
+from ci_tools.functions import discover_targeted_packages
+from ci_tools.parsing import ParsedSetup, parse_require
+
+from pypi_tools.pypi import PyPIClient
+
+try:
+    from jinja2 import Environment, FileSystemLoader
+except:
+    pass  # we only technically require this when outputting the rendered report
+
+
+def get_known_versions(package_name: str) -> List[str]:
+    client = PyPIClient()
+    return client.get_ordered_versions(package_name)
+
+
+def report_should_skip_lib(lib_name: str) -> bool:
+    return "-nspkg" in lib_name
```

Comment:
Out of curiosity, how do we make sure we skip packages we shouldn't check for, now that there's no longer a `skip_pkgs` list (like there was in `analyze_deps.py`)?

Reply:
Otherwise, the normal exclusion by `Inactive` classifier is the way to go.